### PR TITLE
feat(): GIT Integration

### PR DESCRIPTION
**Pull request by marhi**
Added a GITService which checks the current working directory (macos and linux only currently) for a .git folder and displays the current branch.

### CODE REVIEW

File: src/app/services/git.service.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import { Injectable, Provider, EventEmitter, Inject } from '@angular/core';
+import { ConfigService } from './config.service';
+import * as os from 'os';
+import * as fs from 'fs';
+import * as path from 'path';
+let pty = require('node-pty');
+
+export interface Process {
```
Comment (jkuri) [inline src/app/services/git.service.ts]: uneeded I think.

File: src/app/services/git.service.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        return false;
+      }
+    } else {
+      return false;
+    }
+  }
+
+  set branch(b: string) { this.currentBranch = b; }
+  get branch(): string { return this.currentBranch; }
+  set dir(d: string) { this.cwd = d; this.hasGit(); }
```
Comment (jkuri) [inline src/app/services/git.service.ts]: make this in new lines please since you set multiple vars.

File: src/app/services/git.service.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    if (!this.cwd) { return false; }
+    let gitDir: string = path.join(this.cwd, '.git');
+
+    if (gitDir.startsWith('~')) { gitDir = gitDir.replace(/~/, os.homedir()); }
+
+    if (fs.existsSync(gitDir)) {
+      try {
+        let head: string = fs.readFileSync(path.join(gitDir, 'HEAD')).toString();
+        let parts = head.split('/') || null;
+        this.branch = parts[parts.length - 1] || null;
```
Comment (jkuri) [inline src/app/services/git.service.ts]: this now always work unfortunately, for example `pr/81` you get only `81`.
what would be the solution for all cases 
```ts
child_process.exec(`git rev-parse --abbrev-ref HEAD`, (err, stderr, stdour) => {
  if (!err) {
   this.branch = stdout;
  }
});
```

Comment (marhi) [inline src/app/services/git.service.ts]: Thanks, I'll do that and also check for uncommited changes.

**Pull request closed without merging**